### ISSUE TITLE
Missing observations in Twinkles forced sources

### ISSUE THREAD

**Issue by rbiswas4** [DarkEnergyScienceCollaboration/Twinkles#278]
I made this plot of the gband light curve of object ID 3425 using the following code:
https://gist.github.com/rbiswas4/8d67a8b7c8404c349ad38d52bbd7af63
![gband](https://cloud.githubusercontent.com/assets/2281358/16814402/89d927be-48ea-11e6-81cb-96a7bea6ba43.png)

The red points are the ones I obtained from the Twinkles processed data, and the black points are the values that the SN should have had (model truths). The uncertainties in the black points are catsim predictions of flux uncertainties given the m5 values for the night, the uncertainties in red are the uncertainties in DM determination (as interpreted by us, ie. it is possible that we still have a confusion of the correct variables).

1. Why are there three black points with no corresponding red points? This is not that the points are off. If you look at the light curves obtained from the database, it simply has 3 fewer points (9 vs 12)
2. The points near the peak seem to be inferred to have less flux than true. The difference in flux is about 5 percent
3. The uncertainties for these same points seem to be underestimated

**Comment by jchiang87** [comment]
Hi Rahul, I can't look at your notebook on my phone, but I'm wondering if you've checked the forced source FITS file catalogs directly for entries for the missing visits.  At one point, the pserv code omitted entries that had nan values.  I don't think it does that now, but there may be some other issue wrt the content of the dB table vs the catalog files.

**Comment by drphilmarshall** [comment]
Yeah - I wonder, could the missing points be due to:

* failed PhoSim or Stack jobs leading to no image or ForcedSource being
present?
* high background leading to non-detection?

**Comment by rbiswas4** [comment]
@jchiang87 I have not looked at those. Maybe at the meeting tomorrow, someone will point me in a direction as to how to check these.

@drphilmarshall 
- It could be a failed sim/ DM stack job
- I think these are forced photometry results so non-detection should not matter. Now if there are some kind of checks that require the flux to be positive semidefinite, that would cause problems.

**Comment by tony-johnson** [comment]
The 4 visits which were excluded from the DM level 2 processing are listed here: #229

**Comment by rbiswas4** [comment]
Solved the missing data problem: metadata file obtained from @sethdigel and added, The missing data are due to failed phosim runs at ~90 percent moon phase (though the middle of the three) is much less ~66 percent.

**Comment by drphilmarshall** [comment]
Good! Now we know what phosim failures look like in the light curves - its
as if LSST were to breakdown under excessive moonshine  :-/

@rbiswas4 can you please dig out the open issue on these bright time phosim
failures, link it to this issue, and make sure it is labeled a bug and/or
urgent? I think we want to fix this in August, don't you? Thanks!

**Comment by sethdigel** [comment]
These four phosim visits did not fail.  The phosim runs completed in what looks like not excessively long CPU times.  For at least some of them the moon is not even up.  What the visits do have in common is that they are all u band.

**Comment by sethdigel** [comment]
> These four phosim visits did not fail. The phosim runs completed in what looks like not excessively long CPU times. For at least some of them the moon is not even up. What the visits do have in common is that they are all u band.

I should add that I was referring to Tony's four visits listed in #229.  It probably would be remarkable if three of those were responsible for the gaps in the light curve for object ID 3425 above.

**Comment by tony-johnson** [comment]
Right, I only mentioned 4 failed DM visits since we were trying to record all visits that failed. Clearly much more significant are the phosim visits which never finished (110 I think Tom said). Note this it is not per-se a phosim bug (unless you count running slow as a bug) but simply that they exceeded the maximum runtime on the SLAC queues. Also worth noting that the maximum runtime at NERSC is shorter than at SLAC, although perhaps the machines are faster.

**Comment by rbiswas4** [comment]
@sethdigel @tony-johnson 
I should have mentioned in this thread: I checked in the metadata analysis csv into  https://github.com/DarkEnergyScienceCollaboration/Twinkles/tree/master/data 

There is also a notebook now: https://github.com/DarkEnergyScienceCollaboration/Twinkles/blob/master/examples/notebooks/MissingObsHistIds.ipynb which builds the unions of phosim failures and DM stack failures and checks that they add up to all missing observations. Similarly, as you guys guessed from the probability argument of 4/102
all of the missing points on the g band light curve are indeed from failed phosim runs.

@drphilmarshall 
>  can you please dig out the open issue on these bright time phosim
failures, link it to this issue, and make sure it is labeled a bug and/or
urgent? 
Looked at little but was not able to find an issue detailing the phosim time running over limit at SLAC. Did we make an issue in the phosim repository Of course we can get the entire statistics from the metadata analysis
https://github.com/DarkEnergyScienceCollaboration/Twinkles/issues/159

@tony-johnson Agreed, that the failed phosim runs are not exactly bugs but we will always have to deal with limits. I will try to grab the instance catalogs and run phosim on lsstdev (no limit) for the three light curves points that we missed in g band to just see how long it takes. Could you please point me to the override file you finally used?

**Comment by sethdigel** [comment]
Thank you, @rbiswas4.  By my count 103 of the visits failed because the CPU times reached the 120-hour limit on the SLAC batch farm.  These did tend to be visits when the moon was highest in the sky and was the most full.  Of the 103, 61 were g band, 33 were r, and 9 were i.

From Tom's 1sx run (repeating the 1227 visits with 1 s exposure times), I estimated the amount of CPU time that the failed visits would have required, and included that in the column 'cputime_fell' in the metadata file (run1_metadata_v6.csv).  (For visits that did not hit the CPU time limit, that column has the CPU time scaled to the 'fell' class batch hosts.)  The estimated times can be substantial, up to several times the 120-hour limit.

**Comment by tony-johnson** [comment]
Adding @TomGlanzman who should be able to point your to the required override file for the missing phosim runs.

**Comment by rbiswas4** [comment]
@sethdigel Sorry, indeed 103 is the correct number. I was confused about the times when I was trying the older csv file. I updated the notebook with some quantitative description: 
https://github.com/DarkEnergyScienceCollaboration/Twinkles/blob/master/examples/notebooks/MissingObsHistIds.ipynb (SHA : 3d29e53)
Cell 18 confirms the breakdown of bands to be what you said. 
20, and 22 are similar to plots you have made: Essentially, it requires the moon to be above horizon, and the phase to be closer to full. But I think there are also cases where it is the sunalt that is causing the trouble. sunalt > -20.0 (twillight) could be a different issue. If you remember the hack day on OpSim outputs, this was a very significant issue (but it was also one of the cases where OpSim is believed to not be overestimating sky noise).

**Comment by TomGlanzman** [comment]
@rbiswas4 a single override file was used by all Twinkles phoSim runs:

# Turn on debugging file
centroidfile 1

# Does this turn off treerings too?
cleardefects

# Also turn off clouds and airglow variation
clearclouds
airglowvariation 0

# Set the nominal dark sky brightness
zenith_v 21.8

# Leave on CRs but turn off fringing.  ISR will take care of fringing, # but CRs are currently taken out in
# image characterization.
fringing 0

**Comment by johnrpeterson** [comment]
Does this turn off treerings too?

cleardefects

yes






Also turn off clouds and airglow variation

clearclouds
airglowvariation 0

Set the nominal dark sky brightness

zenith_v 21.8

Leave on CRs but turn off fringing. ISR will take care of fringing, # but CRs are currently taken out in
image characterization.

fringing 0
